packages/settingeditor/src/SettingsFormEditor.tsx

```diff
@@ -0,0 +1,207 @@
+/* -----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+import { Settings } from '@jupyterlab/settingregistry';
+import { showDialog } from '@jupyterlab/apputils';
+import React from 'react';
+import Form, { FieldTemplateProps, IChangeEvent, UiSchema } from '@rjsf/core';
+import { JSONSchema7 } from 'json-schema';
+import { JSONExt, ReadonlyPartialJSONObject } from '@lumino/coreutils';
+import { reduce } from '@lumino/algorithm';
+import { PluginList } from './pluginlist';
+import { ISignal } from '@lumino/signaling';
+interface IProps {
+  settings: Settings;
+  renderers: { [id: string]: React.FC };
```

Comment:
Could you make the signature more precise? From rjsf it seems that the type is `React.FunctionComponent<FieldProps> | React.ComponentClass<FieldProps>`
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ce3c84e10e30a3efa6078d5cda57f80c1d539886/types/react-jsonschema-form/index.d.ts#L216